elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java

```diff
@@ -171,10 +172,6 @@ String scriptedFieldPrefix() {
       List<Map.Entry<String, String>> aggregations,
       Long offset, Long fetch) throws IOException {
 
-    if (aggregations.isEmpty()) {
```

Comment:
would you mind explaining a bit why this block could be removed ?

Reply:
Hello, I think if the input has no 'aggregation function' but has 'group by', it should also be handled by aggregate(). Because aggregate() is responsible for interpreting ES aggregation op to request and parsing ES response，a single 'group by' is also ES aggregation.